disdrodb/data_transfer/download_data.py

```diff
@@ -0,0 +1,252 @@
+
+
+def get_yaml_files(metadata_dirs: str) -> list:
+    """Return a list containing all YAML files paths.
+
+
+    Parameters
+    ----------
+    metadata_dirs : str
+        Folder path.
+
+    Returns
+    -------
+    list
+        List of yaml files.
+    """
+
+    return [
+        os.path.join(root, file)
+        for metadata_dir in metadata_dirs
+        for root, dirs, files in os.walk(metadata_dir)
+        for file in files
+        if file.endswith(".yml") or file.endswith(".yaml")
+    ]
+
+
+def get_sation_name_and_url_from_metadata(yaml_file_path: str) -> dict:
+    """Return the station name and the url from the metadata folder.
+
+    Parameters
+    ----------
+    yaml_file_path : str
+        Path to the metadata yaml file.
+
+    Returns
+    -------
+    dict
+        Dictionnary containing the station name and the url.
+    """
+
+    result_dict = {}
+    with open(yaml_file_path) as f:
+        metadata_dict = yaml.safe_load(f)
+
+        result_dict = {k: metadata_dict[k] for k in ["data_url", "station_name"] if k in metadata_dict}
+
+    return result_dict
+
+
+def _get_remote_and_local_data_directories(
+    disdrodb_dir: str = None,
+    data_source: str = None,
+    campaign_name: str = None,
+    station_name: str = None,
+) -> list:
+    """Parse the folder according to the parameters (data_source,
+    campaign_name and station_name) to get all url from the config files.
+
+    Returns
+    -------
+    list
+        List of tuples containing the local path and the url.
+    """
+
+    # raise an error if the path does not ends with "DISDRODB"
+    if not disdrodb_dir.endswith("DISDRODB"):
+        raise ValueError(f"The path {disdrodb_dir} does not end with DISDRODB. Please check the path.")
+
+    # test if the file path is correct and exists. If not, raise an error
+    kwargs = {}
+    if data_source:
+        kwargs["data_source"] = data_source
+    if campaign_name:
+        kwargs["campaign_name"] = campaign_name
+    path_to_analyse = io._get_disdrodb_directory(disdrodb_dir, "RAW", **kwargs)
+
+    # Get all config files from the metadata folders
+    metadata_dir = get_metadata_dirs(path_to_analyse)
+    yaml_files = get_yaml_files(metadata_dir)
+
+    # Get the list of url and local paths from the config files
+    list_of_path_url = []
+    for yaml_file_path in yaml_files:
+        station_name_url = get_sation_name_and_url_from_metadata(yaml_file_path)
+        # Only if the url exists
+        if station_name_url.get("data_url"):
+            # Only if the station name is not specified or if the station name is the same as the config file name
+            if not station_name or station_name == station_name_url.get("station_name"):
+                metadata_dir_path = os.path.dirname(yaml_file_path).replace("metadata", "data")
+                station_dir_path = os.path.join(metadata_dir_path, station_name_url.get("station_name"))

```

Comment:
```suggestion
                station_dir_path = os.path.join(data_dir_path, station_name_url.get("station_name"))
```

Reply:
Also update when the above function will return tuple instead of dict

Reply:
Maybe better: you could modify `get_station_remote_data_url`  to returns directly the (local_dir, remote_url).
```suggestion
                station_dir_path = os.path.join(metadata_dir_path, station_name_url.get("station_name"))
```
At this point you could call the function `get_station_local_remote_path` or something like that